routers/api/v1/repo/pull.go

```diff
@@ -961,6 +961,8 @@ func parseCompareInfo(ctx *context.APIContext, form api.CreatePullRequestOption)
 			return nil, nil, nil, nil, "", ""
 		}
 		headBranch = headInfos[1]
+		// The head repository can also point to the same repo
+		isSameRepo = ctx.Repo.Owner.ID == headUser.ID
```

Comment:
The pre-condition is one user cannot create a fork under his name.

Reply:
I'm sorry, I think I do not understand what does it mean in the context of this change.